code-challenges.js

```diff
@@ -34,6 +80,21 @@ var myNumbers2 = [8, -7, 0, 6, 2]
 
 var coffeeRun = ["Vanilla latte", "Chai tea", "Black coffee", "Mocha", "Americano", "Double espresso"]
 
+//create a function that takes in an array
+const iNeedThis = (array) => {
+    //create an empty array to store odd index elements.
+    //use the filter method to only push elements in odd indexes to a new array
+    let oddIndexes = array.filter((value, index) => {
+        //need more elaboration on why there's an if statement needed if it has one built in, and why it doesn't need curly braces for it to work.
+        if (index % 2 !== 0)
```

Comment:
Let me take a shot at elaborating a bit on how .filter works and why in this example, you'll need an if statement and some curly braces:
Normally, with filter, we're mostly just looking at the VALUE of what we are filtering through, which does our if statement for us. For example, lets consider the following:
`let array = [2, 5, 1, 8, 7, 4]` Here we have an array with random values in each index.

 Now, let's say we want to filter through for only the value and check for even numbers. Our filter can now look something like this: 
`return array.filter(value => value%2 == 0)`  which basically means, if array value at its current index is even, return back the value, else, don't bother returning the value.   When run, we'd receive `[ 2, 8, 4 ]`

So its default, 'built-in' if conditional can be used as is in the above example. 

BUT! In the case of your challenge, we need to do something else though. We want to evaluate the INDEX of the of the array first, THEN do something with the value at the index. 

Because of this, we need to kind of destructure the .filter to get to the index argument first, bypassing the default 'built-in' conditional. We do this by opening up the code block of filter with the curly braces and tell it to access index first, then based upon what we want to evaluate with it, return us the value at the index.

I hope this clears it up a bit! If you have any additional questions, feel free to reach out to me and let me know!